### PR TITLE
Ready for last call. 

### PR DESCRIPTION
**Pull request by gcolvin**
Monday, July 14 would be a good day to start.

### CODE REVIEW

File: EIPS/eip-615.md [hunk 1]
```diff
@@ -1,27 +1,27 @@
 ---
 eip: 615
 title: Subroutines and Static Jumps for the EVM
-status: Draft
+status: Last Call
 type: Standards Track
 category: Core
 author: Greg Colvin <greg@colvin.org>, Brooklyn Zelenka (@expede) , Paweł Bylica (@chfast), Christian Reitwiessner(@chriseth)
-discussions-to: https://ethereum-magicians.org/t/eip-615-subroutines-and-static-jumps-for-the-evm/2728
+discussions-to: https://ethereum-magicians.org/t/eip-615-subroutines-and-static-jumps-for-the-evm-last-call/3472
```
Comment (axic) [inline EIPS/eip-615.md]: Why change the link ?

File: EIPS/eip-615.md [hunk 2]
```diff
@@ -1,27 +1,27 @@
 ---
 eip: 615
 title: Subroutines and Static Jumps for the EVM
-status: Draft
+status: Last Call
```
Comment (axic) [inline EIPS/eip-615.md]: Last call deadline field is missing.

File: EIPS/eip-615.md (hunk 1 above)
Comment (gcolvin) [reply EIPS/eip-615.md]: The original link is 150 comments long, going back to February, so it was getting really hard to follow.

**Comment by axic** [comment]
As a side note, I don't think editors should create branches on this repo.

File: EIPS/eip-615.md [hunk 3]
```diff
@@ -390,7 +391,7 @@ Note that code like this is already run by EVMs to check dynamic jumps, includin
...
+This function can be seen as a symbolic execution of a subroutine in the EVM code, where only the effect of the instructions on the state being validated is computed.  Thus the structure of this function is very similar to an EVM interpreter.  This function can also be seen as an acyclic traversal of the directed graph formed by taking instructions as vertexes and sequential and branching connections as edges, checking conditions along the way.  The traversal is accomplished via recursion, and cycles are broken by returning when a vertex which has already been visited is reached.  The time complexity of this traversal is `O(|E|+|V|): The sum of the number of edges and number of verticies in the graph.
```
Comment (nicksavers) [inline EIPS/eip-615.md]: This removed the formatting. Is this a mistake?

Comment (nicksavers) [inline EIPS/eip-615.md]: First it's _'vertices'_ and then it's _'vertexes'_ and then it's _'verticies'_. I think the first one was correct.

File: EIPS/eip-615.md (hunk 1 above)
Comment (nicksavers) [reply EIPS/eip-615.md]: It's allowed, but difficult for readers to find the whole conversation from start to finished. If somebody in the future wants to understand the reasons why certain decisions were made, they'd need to go to the new last-call discussion and from there find the original discussion.

File: EIPS/eip-615.md (hunk 2 above)
Comment (nicksavers) [reply EIPS/eip-615.md]: `review-period-end:` is missing indeed
According to EIP-1 that's normally 14 days, but I suppose it's up to the author.

File: EIPS/eip-615.md [hunk 4]
```diff
@@ -290,7 +291,7 @@ All of the remaining conditions we validate statically.
 
 #### Costs & Codes
 
-All of the instructions are `O(1)` with a small constant, requiring just a few machine operations each, whereas a `JUMP` or `JUMPI` must do an `O(log n)` binary search of an array of `JUMPDEST` offsets before every jump. With the cost of `JUMPI` being _high_ and the cost of `JUMP` being _mid_, we suggest the cost of `JUMPV` and `JUMPSUBV` should be _mid_, `JUMPSUB` and `JUMPIF` should be _low_, and`JUMPTO` and the rest should be _verylow_.  Measurement will tell.
```
Comment (nicksavers) [inline EIPS/eip-615.md]: Why remove the formatting here? Isn't it better to keep it consistent with the previous one?

File: EIPS/eip-615.md (hunk 1 above)
Comment (gcolvin) [reply EIPS/eip-615.md]: Understood, thus the cross-links.  Not only was the thread already much longer than for any other EIP, but it ended with days of confusing discussion that were really about versioning proposals.  Most of the substantial decisions show up in two threads of earlier discussion from the proposal's years as Issues, also cross-linked, including work with Sydney on a formal specification of the proposal.  A lot of history.  Not, I think, too hard to follow, but more than I'd expect most reviewers to have read.  So it just seemed better to make a clear start to this phase of review.

File: EIPS/eip-615.md (hunk 2 above)
Comment (gcolvin) [reply EIPS/eip-615.md]: I think EIP-1 leaves the date to the discretion of the Editor, not the author.

File: EIPS/eip-615.md (hunk 4 above)
Comment (gcolvin) [reply EIPS/eip-615.md]: No reason, an accident.  Also, Martin informs me, false.  He uses a bitmap to check jump destinations in O(1).

File: EIPS/eip-615.md (hunk 3 above)
Comment (gcolvin) [reply EIPS/eip-615.md]: Yes.  Thanks.

**Comment by gcolvin** [comment]
@nicksavers  On whether changes should delay last call.  Hardly any changes are substantial, and those are fixing small errors.  Mainly I'm clarifying the intro material, (failing) to fix some spelling and formatting, and getting some opcode names right.

**Comment by gcolvin** [comment]
@axic  On branching off this repo.  You are probably right, though it is very convenient, and we do it often enough we should probably discuss when it's appropriate.  I'll continue to use this branch until it's merged.

File: EIPS/eip-615.md (hunk 3 above)
Comment (nicksavers) [reply EIPS/eip-615.md]: The build CI still complains about _'vertexes'_. Even though it's a correct spelling, it's not really common.

Comment (gcolvin) [reply EIPS/eip-615.md]: OK.  Since CI thinks it's a better dictionary than any other I'll obey it ;)

**Comment by nicksavers** [comment]
Note that [EIP-1702](https://eips.ethereum.org/EIPS/eip-1702) has been merged and can be directly linked.

**Comment by gcolvin** [comment]
@axic @nicksavers  I fixed the EIP-1702 link and merged.  Thanks!

File: EIPS/eip-615.md [hunk 5]
```diff
@@ -45,7 +45,7 @@ The result is that all of the following validations and optimizations can be don
 
 ### Dependencies
 
-> **[EIP-1702](https://github.com/ethereum/EIPs/pull/1702). Generalized Account Versioning Scheme.** This proposal needs a versioning scheme to allow for its bytecode (and eventually eWasm bytecode) to be deployed with existing bytecode on the same blockchain.
+> **[EIP-1702](http://eips.ethereum.org/EIPS/eip-1702). Generalized Account Versioning Scheme.** This proposal needs a versioning scheme to allow for its bytecode (and eventually eWasm bytecode) to be deployed with existing bytecode on the same blockchain.
```
Comment (axic) [inline EIPS/eip-615.md]: Btw if this is a dependency, it should have been added to the `requires` header.
